### PR TITLE
[E2E] Rename `.call` to `.call_builder`

### PR DESCRIPTION
**Pull request by cmichi**
## Summary

I always stumble over this:

```rust
let mut call = contract.call::<Flipper>();
let get = call.get();
let get_res = client.call(&ink_e2e::bob(), &get).dry_run().await?;
```

To me this makes more sense:

```rust
let mut call_builder = contract.call_builder::<Flipper>();
let get = call_builder.get();
let get_res = client.call(&ink_e2e::bob(), &get).dry_run().await?;
```

- [y] y/n | Does it introduce breaking changes?
- [n] y/n | Is it dependant on the specific version of `cargo-contract` or `pallet-contracts`?

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have added an entry to `CHANGELOG.md`
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: integration-tests/call-builder-return-value/lib.rs [hunk 1]
```diff
@@ -137,7 +137,8 @@ mod call_builder {
                 .submit()
                 .await
                 .expect("instantiate failed");
-            let mut call_builder_call = call_builder.call::<CallBuilderReturnValue>();
+            let mut call_builder_call =
+                call_builder.call_builder::<CallBuilderReturnValue>();
```
Comment (ascjones) [inline integration-tests/call-builder-return-value/lib.rs:141]: All these instances could do with some renaming to make it clear

Comment (ascjones) [reply integration-tests/call-builder-return-value/lib.rs:141]: Maybe rename the `call_builder` to `contract` like we do in other examples.

**Review by SkymanOne — APPROVED** [review]
LGTM. Just need changes mentioned by Andrew

**Review by smiasojed — APPROVED** [review]

File: integration-tests/call-builder-return-value/lib.rs [hunk 2]
```diff
@@ -137,8 +137,7 @@ mod call_builder {
                 .submit()
                 .await
                 .expect("instantiate failed");
-            let mut call_builder_call =
-                call_builder.call_builder::<CallBuilderReturnValue>();
+            let mut contract = call_builder.call_builder::<CallBuilderReturnValue>();
```
Comment (ascjones) [inline integration-tests/call-builder-return-value/lib.rs:140]:
```suggestion
            let mut call_builder = contract.call_builder::<CallBuilderReturnValue>();
```

And change `let call_builder` on L135 to `let contract = `

**Review by ascjones — APPROVED** [review]

**Pull request merged**